### PR TITLE
fix: improve focus and hover behaviour of reactions menu

### PR DESCRIPTION
**Pull request by arjita-mitra**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - improve accessibility and fix bugs

- The **PR Description**
  - remove unused code
  - close message action menu on tab of last interactive element of a message
  - prevent emoji picker up/down arrow key to navigate between messages in background
  - after emoji selection/esc key then retain the focus on the emoji button
----
##### References
1. https://wearezeta.atlassian.net/browse/SQSERVICES-1734

### CODE REVIEW

File: src/script/components/MessagesList/Message/ContentMessage/MessageActions/MessageReactions/MessageReactionsList.tsx [hunk 1]
```diff
@@ -44,20 +44,23 @@ export interface MessageReactionsListProps {
...
+  onLastReactionKeyEvent,
 }) => {
   const [isSelectedEmoji, setSelected] = useState('');
   const reactionGroupedByUser = groupByReactionUsers(reactions);
   const messageFocusedTabIndex = useMessageFocusedTabIndex(isMessageFocused);
+  const reactionsList = Array.from(reactionGroupedByUser);
   return (
     <div css={messageReactionWrapper}>
-      {Array.from(reactionGroupedByUser).map(([emoji, users], index) => {
+      {Array.from(reactionsList).map(([emoji, users], index) => {
```
Comment (tlebon) [inline src/script/components/MessagesList/Message/ContentMessage/MessageActions/MessageReactions/MessageReactionsList.tsx]: isnt this already an array

Comment (arjita-mitra) [reply src/script/components/MessagesList/Message/ContentMessage/MessageActions/MessageReactions/MessageReactionsList.tsx]: yes, forgot to remove Array.from

File: src/script/components/MessagesList/Message/ContentMessage/MessageActions/MessageReactions/MessageReactions.tsx [hunk 2]
```diff
@@ -66,14 +66,24 @@ const MessageReactions: FC<MessageReactionsProps> = ({
...
+      setShowEmojis(false);
     }
   };
 
+  useEffect(() => {
+    // after emoji selection/esc key press, close the picker and retain the focus on the emoji button
+    if (!showEmojis && currentMsgActionName === MessageActionsId.EMOJI) {
+      if (emojiButtonRef.current) {
+        emojiButtonRef.current.focus();
+      }
```
Comment (przemvs) [inline src/script/components/MessagesList/Message/ContentMessage/MessageActions/MessageReactions/MessageReactions.tsx:83]:
```suggestion
        emojiButtonRef.current?.focus();
```

Comment (arjita-mitra) [reply src/script/components/MessagesList/Message/ContentMessage/MessageActions/MessageReactions/MessageReactions.tsx:83]: i don't think this is required, i have already checked that current is there and the ref is of type HTMLBUTTONELEMENT so focus prop does exist

**Review by tlebon — APPROVED** [review]

**Pull request merged**